src/App.tsx

```diff
@@ -1,93 +1,101 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Filter } from './types/Filter';
+import { Todo } from './types/Todo';
+import { useLocalStorage } from './UseLocalStorage';
+
+const filteredTodos = (todos: Todo[], filter: string) => {
+  const returnArr = [...todos];
+
+  switch (filter) {
+    case Filter.ACTIVE:
+      return returnArr.filter(todo => !todo.completed);
+
+    case Filter.COMPLETED:
+      return returnArr.filter(todo => todo.completed);
+
+    default:
+      return returnArr;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [isChecked, setIsChecked] = useState(false);
+
+  const { filter = '' } = useParams();
+  const visibleTodos = filteredTodos(todos, filter);
+
+  const addTodo = (title: string) => {
+    const newTodo = {
+      id: +new Date(),
+      title,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const deleteTodo = (id: number) => {
+    setTodos(todos.filter(todo => todo.id !== id));
+  };
+
+  const updateTodo = (updatedTodo: Todo) => {
+    setTodos(todos.map(todo => (
+      todo.id === updatedTodo.id ? updatedTodo : todo)));
+  };
+
+  const toggleAll = () => {
+    const allCompleted = todos.every(todo => todo.completed);
+    const updatedTodos = todos.map(todo => ({
+      ...todo,
+      completed: !allCompleted,
+    }));
+
+    setTodos(updatedTodos);
+  };
+
+  const handleChecked = () => {
+    setIsChecked(!isChecked);
+    toggleAll();
+  };
+
+  const clearCompleted = () => {
+    setTodos(todos.filter(({ completed }) => !completed));
+  };
+
   return (
     <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
+      <Header addTodo={addTodo} />
+
+      {todos.length > 0 && (
```

Comment:
```suggestion
      {todos.length && (
```

Reply:
substituted it with  {!!todos.length && (
because  {todos.length && ( adds a '0' when empty list of todos